packages/mwp-api-proxy-plugin/src/util/send.js

```diff
@@ -269,34 +279,33 @@ export function getExternalRequestOpts(request) {
 /**
  * Fake an API request and directly return the stringified mockResponse
  */
-export const makeMockRequest = mockResponse => requestOpts =>
+export const makeMockRequest = (
+	mockResponseContent,
+	responseMeta
+) => requestOpts =>
 	Observable.of([
-		makeMockResponseOk(requestOpts),
-		JSON.stringify(mockResponse),
-	]).do(() => console.log(`MOCKING response to ${requestOpts.url}`));
+		makeMockResponse(requestOpts),
+		JSON.stringify(mockResponseContent),
+	]);
 
 const externalRequest$ = Observable.bindNodeCallback(externalRequest);
 /**
  * Make a real external API request, return response body string
  */
 export const makeExternalApiRequest = request => requestOpts => {
 	return externalRequest$(requestOpts)
-		.do(
-			// log errors
-			null,
-			err => {
+		.catch(err => {
```

Comment:
this is the API error response 'catcher' - instead of bubbling the error up to the parent request handler, it catches and replaces the error with a valid JS object - the application will then have to deal with error data when rendering, but the platform itself will not crash the whole request.

Reply:
Right, so very possible we'll still see errors if the request isn't being handled properly  in components and containers right now?

Reply:
I think this is OK (since things are already broken), just want to confirm.

Reply:
correct - I'm sort of hoping that this will expose some incorrect error-case handling in the apps so that we can find out better ways of managing that. I think there are a _lot_ of places in the app where errors are either ignored or automatically replaced with 'default' values, which isn't ideal.

If I were to re-create the query system, I would structure the error responses differently. Relay has some interesting patterns.